### ISSUE TITLE
Style update: change naming requirements for c++ class members

### ISSUE THREAD

**Issue by lmaisons** [eclipse/omr#45]
Currently our style guide requires that we use leading underscores to identify members in C++ classes.  However there seems to be a community consensus that this isn't such a good idea because:

1. Identifiers containing a leading underscore are reserved implementation use in the global namespace.
2. Identifiers containing a leading underscore followed by a capital letter are universally reserved.
3. Identifiers containing two leading underscores are universally reserved.

Furthermore from the C standard:
>All identifiers that begin with an underscore are always reserved for use as identifiers with file scope in both the ordinary and tag name spaces.

Frequent alternatives seem to be to either trail the identifier with an underscore ```member_``` or to use more than just an underscore in the prefix ```m_member```

[0] http://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier

**Comment by charliegracie** [comment]
I think this is as reasonable change but we have a lot of code that has member names starting with underscores.  I would not want to have to change all of that code but I am ok with stating new code or updated code should follow the `m_member` style

**Comment by alin00** [comment]
+1 @charliegracie's comment.

**Comment by rwy0717** [comment]
Honestly, I don't think we should change our style. I'd rather have consistency in our code. If we have to change, I'm for simply not decorating instance members at all. If we must have something, I vote for the trailing underscore. It's less to type and look at.

How does this rule map to public members and simple structs?

**Comment by keithc-ca** [comment]
I agree that leading underscores aren't ideal, but I'm not a fan of prefixes like 'f_' or 'm_', especially when you have more rules that advise 'g_' or something different for class-static fields. What else would foo be, but a field, in the expression ptr->foo?

**Comment by fjeremic** [comment]
+1 @charliegracie's comment as well. I also agree with @keithc-ca. I disagree with other sentiments that we should keep things as they are for sake of not changing too much code or for consistency. With the introduction of the compiler component our codebase is not consistent to begin with and imposing guidelines for newly contributed code could only benefit us as a project moving forward. Once we settle on a stable guidelines we can refactor existing code.
